internal/execext/exec.go

```diff
@@ -16,17 +16,19 @@ import (
 
 // RunCommandOptions is the options for the RunCommand func
 type RunCommandOptions struct {
-	Command string
-	Dir     string
-	Env     []string
-	Stdin   io.Reader
-	Stdout  io.Writer
-	Stderr  io.Writer
+	Command  string
+	Dir      string
+	Env      []string
+	OneShell bool
+	Stdin    io.Reader
+	Stdout   io.Writer
+	Stderr   io.Writer
 }
 
 var (
 	// ErrNilOptions is returned when a nil options is given
 	ErrNilOptions = errors.New("execext: nil options given")
+    runner *interp.Runner
```

Comment:
Wrong indentation here. Be sure to `go fmt` the code.

Reply:
Hi @andreynering. I corrected the indentation, thanks for pointing that out! I also added documentation, please have a look. The runner is not global, it acts the following way: oneshell command line option is used as default value for all tasks where oneshell parameter is not set explicitly in a taskfile. Please let me know if you see any possible issues in this implementation, thanks.

Reply:
We certainly can't keep `runner` as a global variable here. This would be reused by different tasks and this would cause many different issues: shared state (a `cd` from one task messing with another task, for example), concurrency issues since the library is not thread safe, etc. That's why scoping the runner within each task run / goroutine is needed, even if shared for commands _within the same task_. Some refactor will be needed in the `interp` package to allow that: I think that `interp.RunCommand` will have to be scoped in a struct to keep the runner as shared prop between runs.

Point 2 of my previous comment is still valid as well: I'd keep `oneshell` as a global Taskfile option only (no CLI argument, no task option). If this works well we could promote it to being a default in v4, so the option is just a temporary thing.

Sorry for the amount if info. If you think this is too much, let me know and I can continue the work once I have the time.

Reply:
1) I can implement this behaviour without refactoring mvdan/sh project: create new internal package (interpmanager) with mutex-protected map of interps (for every task). What do you think?
2) I agree that it's reasonable to implement oneshell as a global option if you're planning to make it a default behaviour in v4.